core/gpio/GPIO_digital_driver.h

```diff
@@ -15,15 +15,61 @@
 
 #include <stdint.h>
 #include <string>
+#include <vector>
 
 namespace simba {
 namespace core {
 namespace gpio {
 
+struct Pin{
+  uint16_t number;
+  direction_t direction_;
+};
+class Pins{
+ private:
+  std::vector<Pin> pins;
+
+ public:
+  std::vector<Pin> getPins() {
+  return this->pins;
+  }
+  core::ErrorCode AddPin(uint16_t pinNumber, direction_t direction) {
+    this->pins.push_back(Pin(pinNumber, direction));
+  }
+  bool pinIsRegistered(uint16_t pinNumber) {
+    for (auto pin : this->pins) {
+      if (pin.number == pinNumber) {
+        return true;
+      }
+    }
+    return false;
+  }
+  direction_t getPinDirection(uint16_t pinNumber) {
+    for (auto pin : this->pins) {
+      if (pin.number == pinNumber) {
+        return pin.direction_;
+      }
+    }
+    return direction_t::ERROR;
+  }
+  ErrorCode SetDirection(uint16_t pinNumber, direction_t direction) {
+    for (auto pin : this->pins) {
+      if (pin.number == pinNumber) {
+        pin.direction_ = direction;
+        return ErrorCode::kOk;
+      }
+    }
+    return ErrorCode::kConnectionError;
+  }
+};
+
+
 class GpioDigitalDriver:IgpioDigitalDriver{
  public:
   GpioDigitalDriver();
+  ~GpioDigitalDriver();
 
+  core::ErrorCode initializePin(uint8_t pinNumber, direction_t direction) override;
```

Comment:
to id to jest jakie id. Jak wiemy są 3 już teraz

Reply:
To jest obliczone id GPIO, aplikacja mw dla gpio musi mieć config który będzie je mapował na naszą numerację